crates/bevy_animation/src/curve_path.rs

```diff
@@ -0,0 +1,159 @@
+            let length_after = self.arc_lengths[closest + 1];
+            let segment_length = length_after - length_before;
+
+            let segment_fraction = (distance - length_before) / segment_length;
+
+            (closest as f32 + segment_fraction) / (self.arc_lengths.len() - 1) as f32
+                * self.curve.segments().len() as f32
+        }
+    }
+
+    /// Compute the position of a point on the curve at `distance`.
+    pub fn position(&self, distance: f32) -> P {
+        self.curve.position(self.t(distance))
+    }
+
+    /// Compute the first derivative at `distance`. This is the instantaneous velocity of a point on the cubic curve at `distance`.
+    pub fn velocity(&self, distance: f32) -> P {
+        self.curve.velocity(self.t(distance))
+    }
+
+    /// Compute the second derivative at `distance`. This is the instantaneous acceleration of a point on the cubic curve at `distance`.
+    pub fn acceleration(&self, distance: f32) -> P {
+        self.curve.acceleration(self.t(distance))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::f32::consts::{PI, TAU};
+
+    use bevy_math::{
+        cubic_splines::{CubicBezier, CubicGenerator},
+        vec2, Vec2,
+    };
+
+    use crate::curve_path::CurvePath;
+
+    const ERROR: f32 = 0.001;
+    const SUBDIVISIONS: usize = 256;
+    // The 0.551915 is taken from https://www.mechanicalexpressions.com/explore/geometric-modeling/circle-spline-approximation.pdf
+    const CIRCLE_POINTS: [[Vec2; 4]; 4] = [
+        [
+            vec2(1., 0.),
+            vec2(1., 0.551915),
+            vec2(0.551915, 1.),
+            vec2(0., 1.),
+        ],
+        [
+            vec2(0., 1.),
+            vec2(-0.551915, 1.),
+            vec2(-1., 0.551915),
+            vec2(-1., 0.),
+        ],
+        [
+            vec2(-1., 0.),
+            vec2(-1., -0.551915),
+            vec2(-0.551915, -1.),
+            vec2(0., -1.),
+        ],
+        [
+            vec2(0., -1.),
+            vec2(0.551915, -1.),
+            vec2(1., -0.551915),
+            vec2(1., 0.),
+        ],
+    ];
+
+    #[test]
+    fn length() {
+        let path =
+            CurvePath::from_cubic_curve(CubicBezier::new(CIRCLE_POINTS).to_curve(), SUBDIVISIONS);
+
+        let length = path.length();
+
+        // Length should be 2 PI or TAU
+        assert!(TAU - ERROR < length && length < TAU + ERROR);
+    }
+
+    #[test]
+    fn sample() {
+        let points = [[0.0, 0.2, 0.8, 1.0]];
+        let bezier = CubicBezier::new(points).to_curve();
+        let curve_path = CurvePath::from_cubic_curve(bezier, SUBDIVISIONS);
+
+        assert_eq!(curve_path.length(), 1.0);
+
+        // Check with exact point.
+        assert_eq!(curve_path.t(0.5), 0.5);
+
+        // Check with value between two points.
+        let position = curve_path.position(0.25);
+        assert!(0.25 - ERROR < position && position < 0.25 + ERROR);
+    }
+
+    #[test]
+    fn sample_segmented_curve() {
```

Comment:
Would it be good if this test was a bit more comprehensive and sampled several values (say every 30°?) instead of just `PI`, which is a pretty special value? One could then compare the values with `(cos(distance), sin(distance))` as the "circle" has radius 1. Although we might be running more into issues with the bezier approximation of a circle rather than the sampler itself.